test/util/e2e.go

```diff
@@ -62,3 +67,31 @@ func CreateVisibilityClient(user string) visibilityv1alpha1.VisibilityV1alpha1In
 	visibilityClient := kueueClient.VisibilityV1alpha1()
 	return visibilityClient
 }
+
+func WaitForKueueAvailability(ctx context.Context, k8sClient client.Client) {
+	kcmKey := types.NamespacedName{
+		Namespace: "kueue-system",
+		Name:      "kueue-controller-manager",
+	}
+	deployment := &appsv1.Deployment{}
+	pods := corev1.PodList{}
+	gomega.EventuallyWithOffset(1, func(g gomega.Gomega) error {
+		g.Expect(k8sClient.Get(ctx, kcmKey, deployment)).To(gomega.Succeed())
+		g.Expect(k8sClient.List(ctx, &pods, client.InNamespace("kueue-system"), client.MatchingLabels(deployment.Spec.Selector.MatchLabels))).To(gomega.Succeed())
```

Comment:
Called after ` Eventually` will not shorten the startup wait time when kueue manager is in a crash loop.

We could think of checking  for crashes in after suite ... but I don't see it connected to this issue.

Reply:
> Called after ` Eventually` will not shorten the startup wait time when kueue manager is in a crash loop.

Yes, but it will be better code organization, because it can be called independently after Eventually. Coupling isn't necessary.

> We could think of checking for crashes in after suite ... but I don't see it connected to this issue.

Sure, we can move do it later as we see need. Still, I think it is better to decouple for the ease of reusing later.